pkg/config/vault.go

```diff
@@ -353,7 +353,7 @@ func convertEnvVar(s string) string {
 	if err != nil {
 		log.Entry().Debugf("could not compile regex of convertEnvVar: %v", err)
 	}
-	replacedString := reg.ReplaceAllString(r, "")
+	replacedString := reg.ReplaceAllLiteralString(r, "")
```

Comment:
here it does not really matter, but since we do not expect expansion we should IMO use the `literal` version.